### ISSUE TITLE
Use injected web3.currentProvider

### ISSUE THREAD

**Issue by schmidsi** [paritytech/js-api#6]
I want to use the `web3.currentProvider` injected by MetaMask and others to connect to the Ethereum network.

Similar like this:
```
import Api from "@parity/api";

window.addEventListener("load", () => {
  const api = new Api(window.web3.currentProvider);
  /* ... */
});
```

But it throws the following error:
```
TypeError: this.provider.on is not a function
new PromiseProvider
node_modules/@parity/api/provider/promise.js:24
  21 | super();
  22 | 
  23 | this.provider = provider;
> 24 | this.provider.on('connected', () => this.emit('connected'));
  25 | this.provider.on('connecting', () => this.emit('connecting'));
  26 | this.provider.on('disconnected', () => this.emit('disconnected'));
```

Is this possible somehow? Or are those two incompatible?

**Comment by jacogr** [comment]
Thanks for logging, on the list to do shortly.

**Comment by jacogr** [comment]
Any published version from 2.0.20 (including the current 2.1.1) should allow you to use the `web3.currentProvider` directly. If you find any funnies, let me know.
